### ISSUE TITLE
Ethernet over USB adapter

### ISSUE THREAD

**Issue by vinceHH** [EliasKotlyar/Xiaomi-Dafang-Hacks#236]
Hello,

I need to install the camera in a room where wifi reception is weak. However, this piece has an ethernet connection. Is it possible to connect the camera via a USB ethernet adapter?

thanks !

**Comment by vinceHH** [comment]
hello,

how to add drivers to the firmware ?

thanks

**Comment by jmtatsch** [comment]
Compile them with the toolchain in https://github.com/EliasKotlyar/Xiaomi-Dafang-Software
place them in the driver directory
ismod them in run.sh

**Comment by vinceHH** [comment]
Hello jmtatsch,

thank you for your reply !
I'm happy to hear that it's possible to do it!
my only problem is that I understand the principle of using the toolchain to cross compiler drivers but it lacks some explanations to do it myself.
Could you describe a little more precisely the steps of using the toolchain?

do i have to use one of the compli _ *. sh as an example?

thanks !

**Comment by vinceHH** [comment]
I try to compile the driver by myself but it seems that the makefile include with it used hardcoded path so, it's not working well :(

**Comment by jmtatsch** [comment]
Then fix the hardcoded path?

**Comment by vinceHH** [comment]
Hello,

Sure, it's my goal but it's beyond my skills because I do not know the path to use :(
For information, here the Makefile (renamed with txt extension) included with the driver.

[Makefile.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/1991589/Makefile.txt)

is there a tuto to compile a driver with the toolchain ?

thanks!

**Comment by jmtatsch** [comment]
Have a look at https://github.com/EliasKotlyar/Xiaomi-Dafang-Software/tree/master/drivers for other examples

**Comment by vinceHH** [comment]
Thanks jmtatsch,

I make a try with an example from the drivers but I get an erro from gcc. In fact, I get the same error if I try to compile an existing driver :

```
make[1]: Entering directory '/tmp/Xiaomi-Dafang-Software-master/kernel'

  ERROR: Kernel configuration is invalid.
         include/generated/autoconf.h or include/config/auto.conf are missing.
         Run 'make oldconfig && make prepare' on kernel src to fix it.


  WARNING: Symbol version dump /tmp/Xiaomi-Dafang-Software-master/kernel/Module.symvers
           is missing; modules will have no dependencies and modversions.

  CC [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE/ax88179_178a.o
gcc: error: 0: No such file or directory
gcc: error: unrecognized command line option ‘-G’
gcc: error: unrecognized command line option ‘-mno-abicalls’; did you mean ‘-Wno-finally’?
gcc: error: unrecognized command line option ‘-mlong-calls’
scripts/Makefile.build:308: recipe for target '/tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE/ax88179_178a.o' failed
make[2]: *** [/tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE/ax88179_178a.o] Error 1
Makefile:1224: recipe for target '_module_/tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE' failed
make[1]: *** [_module_/tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE] Error 2
make[1]: Leaving directory '/tmp/Xiaomi-Dafang-Software-master/kernel'
Makefile:22: recipe for target 'modules' failed
make: *** [modules] Error 2
root@PiHole:/tmp/Xiaomi-Dafang-Software-master/drivers/sensors/AX88179_178A_LINUX_DRIVER_v1.19.0_SOURCE#
```

**Comment by nik0** [comment]
@vinceHH,
It is a bit more complicated than that. It seems that the gcc your are using is the local one. You need to "cross" compile: compile for the Dafang Target.
I succeeded compiling the audio driver (but for some reason it crashes when I do the insmod) may be you will have more luck than me with the driver.  
First go in the kernel folder of Xiaomi-Dafang-Software, edit the file `include/config/auto.conf `to change the line `458`
`CONFIG_CROSS_COMPILE="/home/eko/Dokumente/projects/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu`
And set your own path to your own Xiaomi-Dafang-Software
then `make` the kernel !

Once the kernel is compiled,  you can go back to `drivers/audio/oss` (for example) and type `make `(it should work better)
Maybe you will need to modify the makefile for a cross compilation.

 good luck ....

**Comment by vinceHH** [comment]
Hello @nik0,

Thanks for your help !

I'm sorry but I don't find any `include/config` folder in the kernel folder (https://github.com/EliasKotlyar/Xiaomi-Dafang-Software/tree/master/kernel/include)

thank you for your encouragement, I admit to being a newbie on cross compilation (and on linux in fact)

**Comment by nik0** [comment]
My bad this file should have been generated at another moment..
Try this then:
`make CROSS_COMPILE=XXXXXX/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-`

**Comment by vinceHH** [comment]
when I make a try, I get this error :

```
 make CROSS_COMPILE=/tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-
make[1] : on entre dans le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »
/bin/sh: 1: /tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc: not found
/bin/sh: 1: /tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc: not found
make[1]: /tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc : commande introuvable

  ERROR: Kernel configuration is invalid.
         include/generated/autoconf.h or include/config/auto.conf are missing.
         Run 'make oldconfig && make prepare' on kernel src to fix it.
```
So two different error, I don't know why it append "gcc" on the path

So, When I try to make the kernel, I get this error :

 ```
 HOSTLD  scripts/kconfig/conf
scripts/kconfig/conf --silentoldconfig Kconfig
/bin/sh: 1: /home/eko/Dokumente/projects/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc: not found
/bin/sh: 1: /home/eko/Dokumente/projects/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc: not found
```
It speak about /home/eko..., I'm looking for to replace that by my own path

**Comment by nik0** [comment]
Your compiler is not there:
`/tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc`
Try to start it directly, something like this `/tmp/Xiaomi-Dafang-Software/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-gcc`
  

T'es francais ?

**Comment by vinceHH** [comment]
Exact, my bad ... my path is `/tmp/Xiaomi-Dafang-Software-master`

And now, no more gcc erro, thanks !

but if I try to compile the drivers with the original makefile, it seems like I have to customize it :

 ```
make CROSS_COMPILE=/tmp/Xiaomi-Dafang-Software-master/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-
make -C /lib/modules/4.9.0-4-amd64/build SUBDIRS=/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A modules
make[1]: *** /lib/modules/4.9.0-4-amd64/build : Aucun fichier ou dossier de ce type. Arrêt.
Makefile:33 : la recette pour la cible « default » a échouée
make: *** [default] Erreur 2
```


I make a test with an example of a makefile present in https://github.com/EliasKotlyar/Xiaomi-Dafang-Software/tree/master/drivers

It's better but not good:

 ```
make CROSS_COMPILE=/tmp/Xiaomi-Dafang-Software-master/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-
make[1] : on entre dans le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »

  WARNING: Symbol version dump /tmp/Xiaomi-Dafang-Software-master/kernel/Module.symvers
           is missing; modules will have no dependencies and modversions.

  CC [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.o
In file included from include/linux/mmzone.h:18:0,
                 from include/linux/gfp.h:4,
                 from include/linux/kmod.h:22,
                 from include/linux/module.h:13,
                 from /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.c:30:
include/linux/page-flags-layout.h:5:30: fatal error: generated/bounds.h: No such file or directory
compilation terminated.
```

Et oui, je suis français :)

**Comment by nik0** [comment]
Good progress there...
You have to find where is this file "bounds.h"
if you type `find . -name "bounds.h"` you will see that it is in  kernel/include/generated/bounds.h so you have to indicate gcc where to find it. For that you need to use` -IYOURPATH/kernel/include/generated/`

Maybe you can use the command:
`make -IYOURPATH/kernel/include/generated/` or change the makefile to directly add it

tu es d'ou ? Je suis à Bordeaux ...

**Comment by vinceHH** [comment]
:(
there is no bounds.h file on my system 

Je suis un peu plus haut .. du Pas de Calais, dans le nord ;)

**Comment by nik0** [comment]
Strange; I will reset my repo; I keep you posted...

**Comment by vinceHH** [comment]
Got it ... I hope !

I make the kernel which work for a time before stopping for an error :
```
 BC      kernel/timeconst.h
/bin/sh: 1: bc: not found
/tmp/Xiaomi-Dafang-Software-master/kernel/kernel/Makefile:141 : la recette pour la cible « kernel/timeconst.h » a échouée
```

After that, I restart a make of the driver, and it's work !

```
make CROSS_COMPILE=/tmp/Xiaomi-Dafang-Software-master/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-
make[1] : on entre dans le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »

  WARNING: Symbol version dump /tmp/Xiaomi-Dafang-Software-master/kernel/Module.symvers
           is missing; modules will have no dependencies and modversions.

  CC [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.o
  LD [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/sensor_ax88179_178a.o
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/sensor_ax88179_178a.mod.o
  LD [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/sensor_ax88179_178a.ko
make[1] : on quitte le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »
```

Now, I have to wait tonight to test it on my dafang camera

thanks for your help !

**Comment by nik0** [comment]
keep us posted !

**Comment by vinceHH** [comment]
Hello,

:(  that doesn't work :

```
[root@DAFANG-ENTREE:driver]# insmod sensor_ax88179_178a.ko
insmod: can't insert 'sensor_ax88179_178a.ko': unknown symbol in module or invalid parameter
[root@DAFANG-ENTREE:driver]#
```

is the usb port on back of the camera can be used ? did I have to install a driver for the usb port ?

**Comment by vinceHH** [comment]
Hello,

I managed to compile the kernel! it was necessary to install bc.
and now, when I compile the driver with the original makefile, It seems better but I get these warnings:


 ```
make CROSS_COMPILE=/tmp/Xiaomi-Dafang-Software-master/mips-gcc472-glibc216-64bit/bin/mips-linux-gnu-
make -C /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/../../kernel SUBDIRS=/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A modules
make[1] : on entre dans le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »
  CC [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.o
  Building modules, stage 2.
  MODPOST 1 modules
WARNING: "usbnet_set_msglevel" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_get_msglevel" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_tx_timeout" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_start_xmit" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_stop" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_open" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_disconnect" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_probe" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_suspend" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_resume" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_get_endpoints" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_skb_return" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_defer_kevent" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_get_drvinfo" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_write_cmd" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_write_cmd_nopm" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_read_cmd" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
WARNING: "usbnet_read_cmd_nopm" [/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko] undefined!
  CC      /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.mod.o
  LD [M]  /tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A/ax88179_178a.ko
make[1] : on quitte le répertoire « /tmp/Xiaomi-Dafang-Software-master/kernel »
root@debian9:/tmp/Xiaomi-Dafang-Software-master/drivers/AX88179_178A#

```

**Comment by nik0** [comment]
in the driver readme, you should first do a modprobe something. Did you do it ?
Googling your warning I get this https://community.netgear.com/t5/Community-Add-ons/build-external-kernel-moduel/td-p/1052394

**Comment by vinceHH** [comment]
I'm on the same article :)

if I do a modprobe usbnet, I get the error :

```
[root@DAFANG-ENTREE:driver]# modprobe usbnet
modprobe: module 'usbnet' not found
[root@DAFANG-ENTREE:driver]#
```

here my new challenge !

**Comment by nik0** [comment]
In the kernel folder, type: `make menuconfig`
I have to install `sudo apt-get install libncurses-dev`
Then go Device/Drivers -> Network Device Support -> USB Network Adapters
Or edit .config file and enable 
```
CONFIG_USB_USBNET=y
CONFIG_USB_NET_AX88179_178A=y
```
And remake everything, but I don't what will be modified on the kernel and if you need to update something ...

**Comment by vinceHH** [comment]
thanks for your support, I will try .. but after that, I'm don't what I will have to do with this complied kernel !

**Comment by nik0** [comment]
Great well done !

**Comment by jmtatsch** [comment]
Doesn't  `udhcpc -i eth0 -p /var/run/udhcpc.pid -b -x hostname:"$(hostname)"` work?

**Comment by jmtatsch** [comment]
Can you please make a recipe of how to compile a kernel module and create a pull request in the Dafang software repo?

**Comment by hcgonzalezpr** [comment]
Could we get these drivers pre-built on the image?

**Comment by EliasKotlyar** [comment]
You can alway make a pull-request to include any custom software

**Comment by ulflun88** [comment]
This will break with every update until it's included, right?

**Comment by koenvdheuvel** [comment]
Is the dafang or xiaofang able to transfer data over the microusb port?

If so being able to use the chromecast ethernet adapter would be really cool.

What would be even more amazing if someone would build a 802.3af PoE to USB Ethernet + power adapter.

**Comment by EliasKotlyar** [comment]
The Dafang is not able to communicate trough the USB-Port, due to hardware limitation(it only provides power to the dafang). However you can theoretically compile a gadget mode driver, and use it on the normal usb port. It would then act as a usb gadget like "usb network, usb keyboard etc etc"

**Comment by zach-morris** [comment]
@EliasKotlyar , I'm interested in whats required to get the dafang communicating as an ethernet gadget with a connected [Pi Zero setup](https://learn.adafruit.com/turning-your-raspberry-pi-zero-into-a-usb-gadget/ethernet-gadget) as an ethernet gadget.  Looking through the code, are the dafang drivers already included?

**Comment by EliasKotlyar** [comment]
Hello @zach-morris,

Usually not. You will have to compile your own kernel to to so. Start by installing the advanced CFW:
https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/install_sdcard.md

After that, you can try to compile your own kernel and enable the gadget drivers. I suppose that they should work out of the box, like the rest of the hardware.

Greatings
Elias

**Comment by emefbiemef** [comment]
Can we use this method to connect a 5ghz wifi usb dongle like this one? [https://www.aliexpress.com/item/iMice-USB-WiFi-Adapter-PC-600Mbps-AC-Mini-Wireless-Wifi-Antenna-Network-Card-Dual-Band-2/32818267925.html?spm=2114.search0104.3.8.6e1d5d9dnPI027&ws_ab_test=searchweb0_0,searchweb201602_4_10065_10068_319_10059_10884_317_10887_10696_100031_321_322_10084_453_5735415_10083_454_10103_10618_10307_538_537_536,searchweb201603_51,ppcSwitch_0&algo_expid=d6bb5dfc-e602-42b5-ab68-689eca9d60c0-4&algo_pvid=d6bb5dfc-e602-42b5-ab68-689eca9d60c0](url)

**Comment by EliasKotlyar** [comment]
You can use a similar method to compile a driver for this device, and yes, then you can use your camera  with 5GHZ Wifi.

**Comment by emefbiemef** [comment]
I have successfully compiled the driver for [this](https://www.aliexpress.com/item/Wireless-600Mbps-USB-wifi-Adapter-AC600-2-4GHz-5GHz-WiFi-Antenna-PC-Mini-Computer-Network-Card/32848497833.html?spm=2114.search0104.3.71.7b956a47igxoZz&ws_ab_test=searchweb0_0,searchweb201602_1_10065_10068_319_10059_10884_317_10887_10696_100031_321_322_10084_453_10083_454_10103_10618_10307_538_537_536,searchweb201603_51,ppcSwitch_0&algo_expid=d8d6c3ee-120f-4e99-a330-a4495ea0d8d6-9&algo_pvid=d8d6c3ee-120f-4e99-a330-a4495ea0d8d6) usb wifi dongle, using [this](https://github.com/axiomware/RTL8821CU_driver_v5.2.15.3) source.
After `insmod`, it is successfully loaded:
```
[root@10:driver]# insmod 8821cu.ko
[root@10:driver]# lsmod
8821cu 2424628 0 - Live 0xc0899000 (O)
sensor_jxf23 9136 1 - Live 0xc055f000 (O)
tx_isp 363556 3 - Live 0xc04ef000 (O)
sinfo 7814 0 - Live 0xc047e000 (O)
sample_motor 11647 0 - Live 0xc0473000 (O)
audio 60487 0 - Live 0xc0459000 (O)
rtl8189ftv 957549 0 - Live 0xc02c4000 (O)
```
Doing `iwconfig` reveals this:
```
[root@10:driver]# ifconfig wlan1 up
[root@10:driver]# iwconfig
lo        no wireless extensions.

wlan1     unassociated  Nickname:"<WIFI@REALTEK>"
          Mode:Auto  Frequency=2.412 GHz  Access Point: Not-Associated
          Sensitivity:0/0
          Retry:off   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:off
          Link Quality:0  Signal level:0  Noise level:0
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:0   Missed beacon:0

wlan0     IEEE 802.11bgn  ESSID:"NHKNET"  Nickname:"<WIFI@REALTEK>"
          Mode:Managed  Frequency:2.417 GHz  Access Point: BC:EE:7B:C4:C2:50
          Bit Rate:72.2 Mb/s   Sensitivity:0/0
          Retry:off   RTS thr:off   Fragment thr:off
          Encryption key:****-****-****-****-****-****-****-****   Security mode:open
          Power Management:off
          Link Quality=94/100  Signal level=100/100  Noise level=0/100
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:0   Missed beacon:0
```
Scanning also works:
```
[root@10:driver]# iwlist wlan1 scan
wlan1     Scan completed :
          Cell 01 - Address: 2A:C6:3F:E8:31:40
                    ESSID:"DIRECT-iJDESKTOP-9E355JCmsTP"
                    Protocol:IEEE 802.11gn
                    Mode:Master
                    Frequency:2.412 GHz (Channel 1)
                    Encryption key:on
                    Bit Rates:144 Mb/s
                    Extra:rsn_ie=30140100000fac040100000fac040100000fac020c00
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK
                    IE: Unknown: DDBE0050F204104A0001101044000102103B00010010470010E73D15F5253A4A4B948B9093D7BC94AA102100094D6963726F736F66741023000A57696E646F77732050431024000A31302E302E313731333410420001301054000800070050F20000001011000F4445534B544F502D39453335354A431008000200081049000600372A000120104900170001371006001031836E3BC34442258EC9B52AD5818EB71049001B00013720010001052002000F4445534B544F502D39453335354A43
                    Quality=35/100  Signal level=21/100
                    Extra:fm=0003
          Cell 02 - Address: BC:EE:7B:C4:C2:50
                    ESSID:"NHKNET"
                    Protocol:IEEE 802.11bgn
                    Mode:Master
                    Frequency:2.417 GHz (Channel 2)
                    Encryption key:on
                    Bit Rates:144 Mb/s
                    Extra:rsn_ie=30140100000fac040100000fac040100000fac020c00
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK
                    Quality=0/100  Signal level=43/100
                    Extra:fm=0003
          Cell 03 - Address: BC:EE:7B:C4:C2:51
                    ESSID:"NHKNET-Guest"
                    Protocol:IEEE 802.11bgn
                    Mode:Master
                    Frequency:2.417 GHz (Channel 2)
                    Encryption key:on
                    Bit Rates:144 Mb/s
                    Extra:rsn_ie=30140100000fac040100000fac040100000fac020c00
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK
                    Quality=0/100  Signal level=38/100
                    Extra:fm=0003
          Cell 04 - Address: AA:A6:68:C0:3E:E9
                    ESSID:"NazriNHKM_MaxisBroadband"
                    Protocol:IEEE 802.11bgn
                    Mode:Master
                    Frequency:2.442 GHz (Channel 7)
                    Encryption key:on
                    Bit Rates:72 Mb/s
                    Extra:wpa_ie=dd1a0050f20101000050f20202000050f2040050f20201000050f202
                    IE: WPA Version 1
                        Group Cipher : TKIP
                        Pairwise Ciphers (2) : CCMP TKIP
                        Authentication Suites (1) : PSK
                    Extra:
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : TKIP
                        Pairwise Ciphers (2) : CCMP TKIP
                        Authentication Suites (1) : PSK
                    IE: Unknown: DD7E0050F204104A0001101044000102103B00010310470010E6FF93DD854C54AD98391D2414BFFC7D102100035A5445102300084D6564696154656B102400043632383010420004313233341054000800060050F20400011011000F5A544520576972656C65737320415010080002200C103C0001011049000600372A000120
                    Quality=0/100  Signal level=19/100
                    Extra:fm=0003
          Cell 05 - Address: BC:EE:7B:C4:C2:54
                    ESSID:"NHKNET"
                    Protocol:IEEE 802.11an
                    Mode:Master
                    Frequency:5.2 GHz (Channel 40)
                    Encryption key:on
                    Bit Rates:300 Mb/s
                    Extra:rsn_ie=30140100000fac040100000fac040100000fac020c00
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK
                    Quality=72/100  Signal level=43/100
                    Extra:fm=0003
          Cell 06 - Address: BC:EE:7B:C4:C2:55
                    ESSID:"NHKNET-Guest"
                    Protocol:IEEE 802.11an
                    Mode:Master
                    Frequency:5.2 GHz (Channel 40)
                    Encryption key:on
                    Bit Rates:300 Mb/s
                    Extra:rsn_ie=30140100000fac040100000fac040100000fac020c00
                    IE: IEEE 802.11i/WPA2 Version 1
                        Group Cipher : CCMP
                        Pairwise Ciphers (1) : CCMP
                        Authentication Suites (1) : PSK
                    Quality=69/100  Signal level=43/100
                    Extra:fm=0003
```
Now comes the problem, if I try to connect to the AP (NHKNET), it gives this:
```
[root@10:driver]# wpa_supplicant -d -B -i wlan1 -c /system/sdcard/config/wpa_sup
plicant.conf -P /var/run/wpa_supplicant.pid
wpa_supplicant v2.5.nova.1.02
random: Trying to read entropy from /dev/random
Successfully initialized wpa_supplicant
Initializing interface 'wlan1' conf '/system/sdcard/config/wpa_supplicant.conf' driver 'default' ctrl_interface 'N/A' bridge 'N/A'
Configuration file '/system/sdcard/config/wpa_supplicant.conf' -> '/system/sdcard/config/wpa_supplicant.conf'
Reading configuration file '/system/sdcard/config/wpa_supplicant.conf'
ctrl_interface='/var/run/wpa_supplicant'
ctrl_interface_group='0'
ap_scan=1
Priority group 2
   id=0 ssid='NHKNET'
nl80211: Could not add multicast membership for vendor events: -2 (No such file or directory)
rfkill: Cannot open RFKILL control device
nl80211: RFKILL status not available
nl80211: Supported cipher 00-0f-ac:1
nl80211: Supported cipher 00-0f-ac:5
nl80211: Supported cipher 00-0f-ac:2
nl80211: Supported cipher 00-0f-ac:4
nl80211: Disable use_monitor with device_ap_sme since no monitor mode support detected
nl80211: interface wlan1 in phy phy0
nl80211: Set mode ifindex 3 iftype 2 (STATION)
nl80211: Failed to set interface 3 to mode 2: -19 (No such device)
nl80211: Could not configure driver mode
nl80211: deinit ifname=wlan1 disabled_11b_rates=0
nl80211: Remove monitor interface: refcount=0
netlink: Operstate: ifindex=3 linkmode=0 (kernel-control), operstate=6 (IF_OPER_UP)
nl80211: Set mode ifindex 3 iftype 2 (STATION)
nl80211: Failed to set interface 3 to mode 2: -19 (No such device)
wlan1: Failed to initialize driver interface
Failed to add interface wlan1
wlan1: Cancelling scan request
wlan1: Cancelling authentication timeout
```
What do I do now? Please help.

**Comment by ewleonardspock** [comment]
Hi, all.

I was able to get [this](https://www.amazon.com/gp/product/B00M77HMU0/ref=ppx_yo_dt_b_asin_title_o03__o00_s00?ie=UTF8&psc=1) USB Ethernet adapter partially working with the WyzeCam using the ax88179_178a.ko driver and directions from above, but I'm running into an issue where once I run 
`ifconfig eth0 up`
the link lights on the adapter and switch will come on and show some sort of activity, but then after ~ 8 seconds they'll turn back off.  Enabling the ethernet disables WiFi, or at the very least kills my SSH session, and I'm unable to connect to the IP address I have reserved for the adapter's MAC address.  

Any ideas what might be causing this?  Any help is much appreciated :)
